### ISSUE TITLE
損失関数の値が得られない

### ISSUE THREAD

**Issue by saeki-lab** [retarfi/language-pretraining#8]
事前学習モデルとソースコードの提供ありがとうございます。
提供していただいているソースコードを利用して追加学習を試みているのですが、以下のエラーが発生します。

場所：utils.trainer.py 727行目
エラー内容：KeyError 'loss'
![スクリーンショット 2023-02-10 132551](https://user-images.githubusercontent.com/124964579/218000508-eb384abe-e95f-48cd-b8a8-9af7061b036c.png)

outputs.lossの値を確認したところ、結果はNoneでした。
よって、717行目でlossが算出されていないことが原因であると考えていますが、解決できていません。

この問題を解決できることを望んでいます。
よろしくお願いいたします。

**Comment by retarfi** [comment]
ご活用いただきありがとうございます。以下の2点をお聞かせください。
- BERTやELECTRAなど、構築されているアーキテクチャは何になりますでしょうか。
- L.717の`outputs = mode(**inputs)`での`inputs`にはどのkey(`input_ids`, `attention_mask`など)が入力されていますでしょうか。こちらに`labels`がない場合は、`inputs`が正しく生成されていないことになります。

**Comment by saeki-lab** [comment]
ご返信ありがとうございます。
以下、回答になります。

- アーキテクチャは、BERTのsmallになります。
- `inputs`には、`input_ids`、`attention_mask`、`labels`が含まれています。`labels`があるにも関わらず、`loss`が生成されていない状況です。

**Comment by retarfi** [comment]
ご確認ありがとうございます。
BERTの事前学習では、お送りいただいたものの他に、BERTのNext Sentence Prediction(NSP)の正解ラベルに相当する`next_sentence_label`という入力が必要です。
(cf. https://huggingface.co/docs/transformers/model_doc/bert#transformers.BertForPreTraining.forward)
`labels`の他に`next_sentence_label`が入力されていないために`loss`が`None`になると思われます。
バージョンによって入力データセットの作成方法が異なります([v1.0](https://github.com/retarfi/language-pretraining/blob/7d12a21b66d1e540c954ded0288ed808e952d8c6/run_pretraining.py#L91), [v2.0.0](https://github.com/retarfi/language-pretraining/blob/336c3699679dd59be788acc21f83188efa76b95b/create_datasets.py#L174))ので、データセット作成の周辺のコードをご確認いただけますと幸いです。

**Comment by saeki-lab** [comment]
`next_sentence_label`が必要なんですね。
ソースコードを確認して試してみたいと思います。
ご助言ありがとうございます。
